File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> b = new HashSet<Integer>();
        for (int i: banned)
            b.add(i);
        int count = 0;
        for (int i = 1; i < n + 1; i++){
            if (b.contains(i))
                continue;
            maxSum -= i;
            if (maxSum < 0)
                return count;
            count += 1;
        }
        return count;
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        b = set(banned)
        count = 0 
        for i in range(1, n + 1):
            if i in b:
                continue
            
            maxSum -= i
            if maxSum < 0:
                return count
            
            count += 1
        return count


